crates/telio-wg/src/wg.rs

```diff
@@ -388,9 +389,18 @@ impl WireGuard for DynamicWg {
         let _ = self.task.stop().await.resume_unwind();
     }
 
-    async fn reset_existing_connections(&self, exit: PublicKey) -> Result<(), Error> {
+    async fn reset_existing_connections(&self, exit: &ExitNode) -> Result<(), Error> {
+        let pubkey = exit.public_key;
+        let exit_addr = exit.endpoint.map(|addr| addr.ip());
```

Comment:
Damn, I just realized, that we are mixing out-of-tunnel IP addressses and in-tunnel IP addresses here. Got confused myself previously too.

I am thinking, maybe using our own tunnel interface IP address would be better? This means, that setting source IP from endpoint is not really correct :(.

Therefore maybe it would be best to take local tunnel interface IP address (IPv4 one) and use that from this: https://github.com/NordSecurity/libtelio/blob/main/crates/telio-model/src/config.rs#L185 and if this is None (the whole config in `Device::requested_config.meshnet_config`, then default to `10.5.0.2`.

Reply:
I did that and it doesn't work :(

Reply:
After a new round of trying it works finally